src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -955,6 +962,61 @@ public function getPlusType(Expr $left, Expr $right, callable $getTypeCallback):
 			]);
 		}
 
+		if (
+			($leftIsArray->yes() && $rightIsArray->no())
+			|| ($rightIsArray->yes() && $leftIsArray->no())
+		) {
+			return new ErrorType();
+		}
+
+		if (
+			$leftIsArray->yes() && $rightIsArray->maybe()
+		) {
+			if ($rightType instanceof SubtractableType && $rightType->getSubtractedType() !== null) {
```

Comment:
first time I am dealing with a `SubtractableType` in this way. 

I tried leaving this logic to the `TypeCombinator` alone, but couldn't think of a way to express it.

I am curious whether the PR as is is acceptable.